src/app/context/developer_mode.tsx

```diff
@@ -1,35 +1,39 @@
 import {
   AccountManager,
   AztecAddress,
+  AztecAddressLike,
   ExtendedContractData,
   Fr,
   INITIAL_SANDBOX_ENCRYPTION_KEYS,
   INITIAL_SANDBOX_SALTS,
   INITIAL_SANDBOX_SIGNING_KEYS,
   PXE,
   getSchnorrAccount,
+  TxReceipt,
 } from '@aztec/aztec.js';
-import { ReactNode } from 'react';
+import { FieldsOf } from '@aztec/circuits.js';
+import { ReactNode, createContext, useContext } from 'react';
 import { TokenContract } from '../account/token.js';
 import { usePXE } from './pxe.js';
 import { useMutation, useQuery } from '@tanstack/react-query';
 import { InfoDialog } from '../modals/InfoDialog.js';
+import { useTokenList } from './token_list.js';
 
 function getSandboxAccounts(pxe: PXE): AccountManager[] {
   return INITIAL_SANDBOX_ENCRYPTION_KEYS.map((encryptionKey, i) =>
     getSchnorrAccount(pxe, encryptionKey, INITIAL_SANDBOX_SIGNING_KEYS[i], INITIAL_SANDBOX_SALTS[i]),
   );
 }
 
-const TOKEN_CONTRACT_ADDRESS = '0x2f45f498b7912c779dde8e3594622e36d7908088b09e99ab91caaafb40d1f9ef';
 
 export function DeveloperModeProvider({ children }: { children: ReactNode }) {
   const { pxe } = usePXE();
   const adminAccount = getSandboxAccounts(pxe)[0];
+  const tokenContractAddress = useTokenList()[0].address;
```

Comment:
This is minor but I would just use `TOKEN_LIST[0].address`, I don't see the advantage of using hooks for constants. Hooks implies that there is mutable state underneath. But it doesn't bother me much.

Reply:
I assume in the future there will be (mutable?) state, i.e. list will be stored on chain, or have properties that are fetched from chain.

Reply:
✅ Ok, you convinced me, done.